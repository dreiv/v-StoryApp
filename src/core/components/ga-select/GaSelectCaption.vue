<script setup lang="ts">
defineProps<{ title?: string }>()
</script>

<template>
  <caption :class="$style.caption">
    <slot name="title">{{ title }}</slot>
  </caption>
</template>

<style module>
.caption {
  display: flex;
  align-items: center;

  padding: 0 var(--ga-size-spacing-04);
  height: 2.25rem; /* TODO: fix */

  font-weight: var(--ga-font-weight-semibold);
  font-size: var(--ga-text-sm-font-size);
  line-height: var(--ga-text-sm-line-height);
  letter-spacing: var(--ga-text-sm-tracking);
}
</style>
