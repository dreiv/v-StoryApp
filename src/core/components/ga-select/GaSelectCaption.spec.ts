import { shallowMount, type VueWrapper } from '@vue/test-utils'
import GaSelectCaption from './GaSelectCaption.vue'

describe('GaSelectCaption', () => {
  let wrapper: VueWrapper

  const createWrapper = (propsData = {}, slots = {}) => {
    return shallowMount(GaSelectCaption, {
      props: propsData,
      slots,
    })
  }

  it('should render correctly with title prop', () => {
    wrapper = createWrapper({ title: 'My Caption Title' })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.text()).toContain('My Caption Title')
    expect(wrapper.classes()).toContain('caption')
  })

  it('should render title slot content instead of title prop if provided', () => {
    wrapper = createWrapper(
      { title: 'Prop Title' },
      { title: '<span class="custom-caption-slot">Slot Title</span>' },
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.custom-caption-slot').exists()).toBe(true)
    expect(wrapper.text()).toContain('Slot Title')
    expect(wrapper.text()).not.toContain('Prop Title')
  })

  it('should render with default slot if no title prop or slot is provided', () => {
    wrapper = createWrapper()
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.text()).toBe('')
  })
})
