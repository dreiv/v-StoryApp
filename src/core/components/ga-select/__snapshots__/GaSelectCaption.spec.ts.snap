// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GaSelectCaption > should render correctly with title prop 1`] = `"<caption class="caption">My Caption Title</caption>"`;

exports[`GaSelectCaption > should render title slot content instead of title prop if provided 1`] = `"<caption class="caption"><span class="custom-caption-slot">Slot Title</span></caption>"`;

exports[`GaSelectCaption > should render with default slot if no title prop or slot is provided 1`] = `"<caption class="caption"></caption>"`;
